src/app/asset-page/asset-page.component.ts

```diff
@@ -321,13 +321,15 @@ export class AssetPage implements OnInit, OnDestroy {
                 }
 
                 // For "Go Back to Results" and pagination, for asset that is not from image group look for requestId to set prevRouteParams
-                if (routeParams['requestId'] || routeParams['groupId']) {
+                if (routeParams['requestId'] || routeParams['groupId'] || routeParams['prevRouteTS']) {
```

Comment:
Rather than using the timestamp alongside requestId, let's replace requestId's usage. And we should also clean out the "groupId" condition here